I've got Your other patch checked out on my sandbox and my hosts files is updated. I pulled up this PR locally and I'm seeing a JS error when I try and go to a prompt tag page:

`caught (in promise) Error: Cannot find module 'calypso/state/reader/tags/selectors/get-reader-tag-by-slug'`

![CleanShot 2023-05-08 at 14 51 12](https://user-images.githubusercontent.com/5634774/236907636-8fa460aa-974c-4c5f-b80b-78304ceac8cd.png)

which prevents the page from loading.

--- a/client/reader/tag-stream/header.jsx
+++ b/client/reader/tag-stream/header.jsx
@@ -19,10 +19,16 @@ class TagStreamHeader extends Component {
 	render() {
 		const { title, description, isPlaceholder, showFollow, following, onFollowToggle, showBack } =
 			this.props;
+
+		// A bit of a hack: check for a prompt tag (which always have a description) from the slug before waiting for tag info to load,
+		// so we can set a smaller title size and prevent it from resizing as the page loads. Should be refactored if tag descriptions
+		// end up getting used for other things besides prompt tags.
+		const isPromptTag = new RegExp( /^dailyprompt-\d+$/ ).test( title );
+
 		const classes = classnames( {
 			'tag-stream__header': true,
 			'is-placeholder': isPlaceholder,
-			'has-description': description,
+			'has-description': isPromptTag || description,
 			'has-back-button': showBack,
 		} );
 

--- a/client/reader/tag-stream/header.jsx
+++ b/client/reader/tag-stream/header.jsx
@@ -1,5 +1,5 @@
 import classnames from 'classnames';
-import { localize } from 'i18n-calypso';
+import { localize, translate } from 'i18n-calypso';
 import PropTypes from 'prop-types';
 import { Component } from 'react';
 import { connect } from 'react-redux';
@@ -17,26 +17,21 @@ class TagStreamHeader extends Component {
 	};
 
 	render() {
-		const {
-			title,
-			isPlaceholder,
-			hasLongTitle,
-			showFollow,
-			following,
-			onFollowToggle,
-			translate,
-			showBack,
-		} = this.props;
+		const { title, description, isPlaceholder, showFollow, following, onFollowToggle, showBack } =
+			this.props;
 		const classes = classnames( {
 			'tag-stream__header': true,
 			'is-placeholder': isPlaceholder,
-			'has-long-title': hasLongTitle,
+			'has-description': description,
 			'has-back-button': showBack,
 		} );
 
 		return (
 			<div className={ classes }>
-				<h1 className="tag-stream__header-title">{ title }</h1>
+				<div className="tag-stream__header-title-group">
+					<h1 className="tag-stream__header-title">{ title }</h1>
+					{ description && <h2 className="tag-stream__header-description">{ description }</h2> }
+				</div>
 				<div className="tag-stream__header-follow">
 					{ showFollow && (
 						<FollowButton

--- a/client/reader/tag-stream/main.jsx
+++ b/client/reader/tag-stream/main.jsx
@@ -45,10 +45,11 @@ class TagStream extends Component {
 			}
 		} );
 		asyncRequire( 'twemoji', function ( twemoji ) {
+			const title = self.props.decodedTagSlug;
 			if ( self._isMounted ) {
 				self.setState( {
 					twemoji,
-					isEmojiTitle: self.props.title && twemoji.test( self.props.title ),
+					isEmojiTitle: title && twemoji.test( title ),
 				} );
 			}
 		} );
@@ -99,13 +100,14 @@ class TagStream extends Component {
 
 	render() {
 		const emptyContent = <EmptyContent decodedTagSlug={ this.props.decodedTagSlug } />;
+		const title = this.props.decodedTagSlug;
 		const tag = find( this.props.tags, { slug: this.props.encodedTagSlug } );
 
 		let imageSearchString = this.props.encodedTagSlug;
 
 		// If the tag contains emoji, convert to text equivalent
 		if ( this.state.emojiText && this.state.isEmojiTitle ) {
-			imageSearchString = this.state.emojiText.convert( this.props.title, {
+			imageSearchString = this.state.emojiText.convert( title, {
 				delimiter: '',
 			} );
 		}
@@ -117,7 +119,7 @@ class TagStream extends Component {
 					<QueryReaderTag tag={ this.props.decodedTagSlug } />
 					{ this.props.showBack && <HeaderBack /> }
 					<TagStreamHeader
-						title={ this.props.title }
+						title={ title }
 						imageSearchString={ imageSearchString }
 						showFollow={ false }
 						showBack={ this.props.showBack }
@@ -130,7 +132,7 @@ class TagStream extends Component {
 		return (
 			<Stream
 				{ ...this.props }
-				listName={ this.props.title }
+				listName={ title }
 				emptyContent={ emptyContent }
 				showFollowInHeader={ true }
 				forcePlaceholders={ ! tag } // if tag has not loaded yet, then make everything a placeholder
@@ -139,23 +141,19 @@ class TagStream extends Component {
 				<QueryReaderTag tag={ this.props.decodedTagSlug } />
 				<DocumentHead
 					title={ this.props.translate( '%s ‹ Reader', {
-						args: this.props.title,
+						args: title,
 						comment: '%s is the section name. For example: "My Likes"',
 					} ) }
 				/>
 				{ this.props.showBack && <HeaderBack /> }
 				<TagStreamHeader
-					title={
-						this.props.isPromptTag && this.props.description
-							? this.props.description
-							: this.props.title
-					}
+					title={ title }
+					description={ this.props.description }
 					imageSearchString={ imageSearchString }
 					showFollow={ !! ( tag && tag.id ) }
 					following={ this.isSubscribed() }
 					onFollowToggle={ this.toggleFollowing }
 					showBack={ this.props.showBack }
-					hasLongTitle={ this.props.isPromptTag }
 				/>
 			</Stream>
 		);
@@ -169,9 +167,7 @@ export default connect(
 			description: tag?.description,
 			followedTags: getReaderFollowedTags( state ),
 			tags: getReaderTags( state ),
-			title: tag?.displayName || decodedTagSlug,
 			isLoggedIn: isUserLoggedIn( state ),
-			isPromptTag: new RegExp( /^dailyprompt-\d+$/ ).test( decodedTagSlug ),
 		};
 	},
 	{

--- a/client/reader/tag-stream/style.scss
+++ b/client/reader/tag-stream/style.scss
@@ -6,14 +6,26 @@
 	margin-bottom: 10px;
 }
 
+.tag-stream__header-title-group {
+	display: flex;
+	flex-wrap: wrap;
+	margin-right: 6px;
+}
+
 .tag-stream__header-title {
 	font-size: 2rem;
 	text-transform: capitalize;
 	font-weight: 600;
+	margin-bottom: 4px;
 }
 
-.has-long-title .tag-stream__header-title {
-	font-size: 1.5rem;
+.has-description .tag-stream__header-title {
+	font-size: 1.25rem;
+}
+
+.tag-stream__header-description {
+	font-size: 2rem;
+	font-weight: 600;
 }
 
 // Back button in Tag streams

--- a/client/state/reader/tags/selectors/get-reader-tag-by-slug.js
+++ b/client/state/reader/tags/selectors/get-reader-tag-by-slug.js
@@ -1,0 +1,14 @@
+import 'calypso/state/reader/init';
+
+/**
+ * Returns all of the reader tags cached in calypso
+ *
+ *
+ * @param {Object}  state  Global state tree
+ * @returns {Array}        Reader Tags
+ */
+
+export default function getReaderTagBySlug( state, slug ) {
+	const tags = state.reader.tags.items;
+	return tags ? Object.values( tags ).find( ( tag ) => tag.slug === slug ) : null;
+}

